redfish-core/lib/assembly.hpp

```diff
@@ -365,9 +408,106 @@ inline void setAssemblylocationIndicators(
         {
             setLocationIndicatorActive(asyncResp, assembly, iter->second);
         }
+
+        // Handle special case for tod_battery assembly OEM ReadyToRemove property
+        // NOTE: The following method for the special case of the tod_battery
+        // ReadyToRemove property only works when there is only ONE adcsensor
+        // handled by the adcsensor application.
+        if (sdbusplus::message::object_path(assembly).filename() == "tod_battery")
+        {
+            if (!json_util::readJson(req, asyncResp->res, "Assemblies", Assemblies, "Oem", oem)) 
+            {
+                const nlohmann::json& thisJson = *Assemblies;
```

Comment:
Why do you need this `nlohmann::json& thisJson`? 
Why doesn't assemblyData just work?